pyogrio/_err.pyx

```diff
@@ -187,40 +234,45 @@ cdef void *exc_wrap_pointer(void *ptr) except NULL:
     return ptr
 
 
-cdef int exc_wrap_int(int err) except -1:
-    """Wrap a GDAL/OGR function that returns CPLErr or OGRErr (int)
-    Raises an exception if a non-fatal error has be set.
+cdef int check_int(int err) except -1:
+    """Check the CPLErr (int) value returned by a GDAL/OGR function.
 
-    Copied from Fiona (_err.pyx).
+    If `err` is a nonzero value, an exception inheriting from CPLE_BaseError is
+    raised.
+    When the last error registered by GDAL/OGR was a non-fatal error, the
+    exception raised will be customized appropriately. Otherwise a CPLError is
+    raised.
     """
     if err:
-        exc = exc_check()
+        exc = check_last_error()
         if exc:
             raise exc
         else:
             # no error message from GDAL
-            raise CPLE_BaseError(-1, -1, "Unspecified OGR / GDAL error")
+            raise CPLError(-1, -1, "Unspecified OGR / GDAL error")
+
     return err
 
 
-cdef int exc_wrap_ogrerr(int err) except -1:
-    """Wrap a function that returns OGRErr (int) but does not use the
-    CPL error stack.
+cdef int check_ogrerr(int err) except -1:
```

Comment:
Maybe for another PR, but I am wondering if we should keep this helper function. 
At the moment we use this for calling `OGR_L_GetExtent`, but then if the return value is non-zero, that indicates the bounds are not known and we catch the error to set the bounds to `None`. But so that checking for non-zero return we can do directly where we call `OGR_L_GetExtent`, instead of raising a python error and catching that again to do the same.

And the other place it is used is in `exc_wrap_ogrerr(exc_wrap_int(OSRImportFromEPSG(srs, 4326)))`, which seems a bit fishy in that it chains both exception handlers. I don't think the `exc_wrap_ogrerr` has any effect there.

Reply:
In both cases the added value is indeed not very obvious...

Reply:
I'll remove it.